### DISCUSSION TITLE
Add Google Wear app for Liftosaur

### DISCUSSION THREAD

**Discussion by astashov** [astashov/liftosaur#118] [Ideas]
So you could do from your watch:

* Switch between exercises of the current workout
* Mark a set of any exercise of the current workout, including amraps
* Edit a set of any exercise of the current workout
* See rest timer
* Adjust rest timer
* Get notifications when rest timer is firing

**Comment by Schnibbz** [comment]
This absolutely NEEDS to happen.  I have been checking in on progress periodically and am happy to see it listed here now.

**Reply by BigHunanBalls** [reply]
Agreed this is necessary for any lifting app

**Reply by Liam41** [reply]
Completely Agreed! It would be so nice to be able to just look at my watch. for Reps / weight needed / plate math and any timer involved. I dont honestly need even sync to health connect or metrics!

**Comment by Petrochempros** [comment]
Watch implementation would take this app to the next level

**Comment by hubmacfan** [comment]
I could not vote for this feature enough!  Honestly, this is what keeps me from switching from using Hevy and a spreadsheet to Liftosaur.

**Comment by slanov** [comment]
I'm just testing the app at the moment. I like what I see so far, but the lack of an Apple Watch integration makes the migration from Strong + Excel sheet difficult.

If an Apple Watch version was included, I would purchase premium immediately.

**Comment by jayquez** [comment]
I moved from StrongLifts to this app, bought the lifetime pass. This is the biggest feature I’m missing from StrongLifts.

**Reply by Schnibbz** [reply]
I don’t think Anton gives a fuck. He started this GitHub to gauge interest but prefers to ignore make micro-tweaks to his coding than code in the most important feature to the app. Seriously, SL 5x5 app has watch support. What the actual fuck? It’s also cheap as hell! Might be easier to  ask Mehdi to program in GZCL into his app then we could drop our sub to Liftosaur.

**Reply by jayquez** [reply]
Yeah but the flexibility of Liftosaur in everything else is unmatched.

**Reply by AdEmergency625** [reply]
I agreed the flexibility is bar none and smoke everything out there. I could see people who don’t see values in the programming features improvements. From my personal experience, Anton had addressed so many of my issues accurately and quickly even before I pay for the app. Yes, the app version didn’t give me much more than the free version. But I’m willing to invest my money to help the app to improve and wait for useful features like Apple Watch. 

I’m in IT field. Asking for a feature is super easy. Developing it bug free is way harder than many realized especially the number of devices, OS and various combinations. And companies like Apple keeps changing how they implement the platform too. 

Disclaimer: I have no personal relationship with Anton or this company. I only speak out of my experience honestly.

**Reply by astashov** [reply]
> I don’t think Anton gives a fuck

I definitely do

**Reply by Schnibbz** [reply]
Then stop ignoring your subscribers; this post was a good start, but it’s frustrating that you skip over discord questions you’re tagged on about watch support. It’s not like I’m hammering you on it, it had been at least 5 or 6 weeks since I last asked when I recently posted in discord. You responded to 2 or 3 other tagged comments above mine and skipped the watch question. 

You also said a year ago that it was in the pipeline for late 2024. I didn’t like hearing it then but it’s now late 2024, so let’s giddy up!

I get it’s a big undertaking but you’re generating revenue with subscribers, go hire a coder from India or Pakistan, like so many small devs do! 

I’ve canceled my subscription which was set to renew in 6 weeks or so. I won’t be purchasing again until I see that there is at least some movement on watch implementation.

**Reply by astashov** [reply]
> You responded to 2 or 3 other tagged comments above mine and skipped the watch question.

I'm really sorry about that, I try to answer everything on Discord, but probably missed that one.

I understand it could be frustrating to wait, but Apple Watch is essentially a new app - it's a large multi-month project, so I'm trying to balance ROI on various features, taking into account how much effort it takes to implement the feature and how many people request it. E.g. Apple Health/Google Health Connect sync is on the second place, but it's way less effort - so I'm working on that now. 

Apple Watch is definitely the most requested, but also the most complex. First I need to learn how to build watch apps, learn a new language, etc - so takes a while. Outsourcing that is not that easy - good developers cost a lot (I know - I definitely couldn't afford to hire myself :)), so I'd have to do it myself.

I'm sorry to see you go, but check other tracker apps, like Hevy or Alpha Progression - if you don't really need advanced programming capabilities or automation, but value the watch support more - they could be a good fit for you.

**Reply by Liam41** [reply]
I cant speak for everyone but would honestly say that ROI on an apple watch / wear os watch app implementation is going to provide more ROI than health connect. Most people that I talk to just start a "workout" with samsung health / apple health / google fit and then run their weight tracking app at the same time. The health apps will provide all the heart monitoring and calorie information much better than the sync to health connect from liftosaur will ever be able to provide anyways (not trying to be critical, but just a fact that its complicated stuff)

I think if you can get the watch apps built to just show the following and forget health connect syncing all together, I think its truly a better ROI in the long term.
1. Switch between exercises of the current workout
2. Mark a set of any exercise of the current workout, including amraps
3. Edit a set of any exercise of the current workout
4. See rest timer
5. Adjust rest timer
6. Get notifications when rest timer is firing
7. See plate math
8. see expected reps / weight

I honestly do think that health connect in any of these apps is useless because it just inputs a time of workouts into the health apps and estimates calories and such which is pointless for an app like this. If anything it would be better to figure out a way to trigger the health apps to start recording a workout simultaneously! (Just my honest thoughts)

**Comment by AdEmergency625** [comment]
Yeaaaaaahhhh! I knew you would listen to this request. If you’re looking for best practices, check out Streaks. In general Streaks app has one of the best Apple implementations and integration. Simple interface, easy to use and awesome integration with Apple Watch, IPhone, iPad, Mac, Apple Health and Apple shortcuts.

**Reply by Petrochempros** [reply]
Well to be fair he hasn't listened yet this has yet to be started. Needs to be done immediately

**Comment by TimKamppinen** [comment]
I must be the only person who deliberately removes my apple watch before lifting, lol. For me it just gets in the way and would much rather just use the phone interface than trying to fumble with the little dial and tiny touchscreen.

**Reply by DrPeprZero** [reply]
Yeah, no watch or jewelry for me either when I'm lifting. They just get in the way.
I'd only consider a wearable if it could track all my lifts (sets/reps etc) automatically and without any interaction. Ofc. that's a utopian idea.

**Comment by alecmev** [comment]
Not sure how common it is, but I personally stick to Strong because I want to see my heart rate, as it’s what I usually base my decision to do the next set on. Timers are only “reminders”, for me to check my HR. I also like that Strong creates a workout in Health and estimates the energy expenditure. The rest is much less pressing, as it can be done on a phone just fine.

**2025-05-18 edit**: Take a look at Strong's 6.0 Apple Watch rewrite for inspiration, great UX.

**Reply by John-Gee** [reply]
Oh that's interesting I never thought of that. I'll have to try! :)

**Reply by amine250** [reply]
At which HR you decide to go on to the next set ?

**Reply by alecmev** [reply]
Around 55% HRmax, but it's not scientific, just a threshold below which I haven't noticed myself struggle.

**Comment by arnabarnabarnab** [comment]
I’m already a paid-up user, but would 100% pay more for this feature.

**Comment by Schnibbz** [comment]
what happened to Apple Watch support being the next issue to be tackled? Seems like there hasn’t been much said since it was announced as being on the horizon.Sent from my iPhoneOn Nov 3, 2025, at 1:35 PM, arnabarnabarnab ***@***.***> wrote:﻿
I’m already a paid-up user, but would 100% pay more for this feature.

**Comment by stjema** [comment]
watches cant track kcal or anything.

**Comment by astashov** [comment]
Apple Watch is done

**Reply by alecmev** [reply]
Thank you! 🙏🏼

**Reply by avis911** [reply]
> Apple

Feedback on current watch UI, coming from StrongLifts

I recently switched from StrongLifts 5x5 to Liftosaur (StrongLifts dropped its lifetime license) and have been really happy with the programmability — being able to script my own progression logic is a huge upgrade. The one thing I miss is how simple StrongLifts' watch screen was: big weight, big reps, one tap to confirm, done.

Right now during a set, my watch shows quite a bit at once — elapsed time, heart rate, set counter, target reps × weight (often to 2 decimal places due to plate rounding), plus a warmup line and a plates breakdown, all on one screen. Mid-set, glancing down, it takes a second to find the one number I actually need (what to lift right now).

A "simple mode" toggle would be great — something like:

Big, single-glance weight × reps for the current set
One tap to confirm
Everything else (warmup detail, plate breakdown, HR, timer) tucked behind a swipe or a details view, not shown by default

Totally understand this is a big ask for a solo dev, and I know the current watch experience is already flagged as a known gap in this thread — just wanted to add a concrete example of what "simpler" could look like in practice, using StrongLifts as the reference point. Happy to help test if this ever gets prioritized.

**Reply by avis911** [reply]
But a BIIIIIG Thank you for the Apple Watch support

**Reply by avis911** [reply]
The Heart rate is fantastic, I think the rest timer could have its own screen with just heart rate and a countdown timer that one taps to go back into the lift part, I often use the heart rate to judge how quickly I am recovering from a set, it helps with gauging todays fitness level, as not every day is the same. Just an idea for you and thanks again for all your work on this.

**Reply by astashov** [reply]
Everything on a single screen is kinda a personal preference, sorry :) I like when I can see everything at once - especially plates, target and the current rest timer...

**Reply by avis911** [reply]
Generally you are right it’s better to have everything at one place , unfortunately the watch screen and I mean only the watch screens have  very little real estate, I am almost 50 and sometimes it’s a bit of struggle to see smaller font stuff , especially when color contrasts are a bit muted. Maybe larger metrics all together on the phone and just fewer things on display on the watch ( core essentials) Just my 2 cents ! But I love what you are doing with it overall ! The app is stable and has not frozen or crashed even once ! It syncs to the phone easily and I love the ability to program my own metrics using LLM. I will stick with it for sure.

**Comment by palenshus** [comment]
Any word on Android watch support?
